### PR TITLE
Add engine schema validation

### LINKED ISSUE #16856: Engine - Validate schema fields and types on component build time
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
|5.0.0 | Engine | Manager | Packages/Sources | OS version |

Currently, the engine does not ensure that mappings on schema fields are compliant with the schema definition.

We need to check in build time if fields mapped on components comply with the schema definition of said field so integrity on the indexer is maintained while giving the user feedback on errors.

### PR DESCRIPTION
**Pull request by JavierBejMen**
|Related issue|
|---|
| Closes #16856 |

The solution presented in this PR moves forward regarding the integrity of the indexer and improves user feedback on schema-related errors.

Here is a brief overview of the changes implemented:

- **Schema Management Module**: A foundational module has been added to manage the schema. This module is the bedrock for the new schema verification feature and forms the basis for the subsequent improvements.

- **Schema Loading and Verification**: Schema loading from JSON descriptions has been implemented, alongside schema verification on basic operations. This ensures that any interactions with the schema are validated against the definitions, enhancing data integrity and reducing the risk of errors.

- **Field Naming Validation**: Naming validation has been added. This addition will further safeguard against potential inconsistencies or errors in the schema, ensuring that field naming adheres strictly to the schema definition or custom fields.

- **Updated Ruleset and Tooling**: Finally, the ruleset and tooling have been updated to work seamlessly with the new schema verification feature.

### CODE REVIEW

File: src/engine/test/source/schemf/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+add_executable(schema_test schema_test.cpp)
+target_link_libraries(schema_test gtest_main schemf)
+gtest_discover_tests(schema_test)
+
+add_executable(field_test field_test.cpp)
+target_link_libraries(field_test gtest_main schemf)
+# gtest_discover_tests(field_test)
```
Comment (LucioDonda) [inline src/engine/test/source/schemf/CMakeLists.txt:7]: shouldn't this be uncommented?

**Review by LucioDonda — CHANGES_REQUESTED** [review]

File: src/engine/source/builder/builders/operationBuilder.cpp [hunk 2]
```diff
@@ -108,56 +139,88 @@ enum class OperationType
...
 
-    std::string field;
+    if (schema == nullptr)
+    {
+        throw std::runtime_error("operation builder needs a valid schema.");
+    }
+
+    std::string targetField;
+    std::string targetFieldPath;
+    std::string operationName;
```
[the remark this answers is not in the document]
Comment (JavierBejMen) [reply src/engine/source/builder/builders/operationBuilder.cpp:158]: Initialization of the variables needs to be safe guarded for exceptions, to make it const requires excessive boilerplate

File: src/engine/test/source/schemf/CMakeLists.txt (hunk 1 above)
Comment (JavierBejMen) [reply src/engine/test/source/schemf/CMakeLists.txt:7]: My mistake, done.

**Review by LucioDonda — APPROVED** [review]
LGTM!

**Review by nbertoldo — APPROVED** [review]
LGTM!

**Review by JcabreraC — APPROVED** [review]
LGTM !

**Pull request merged**